Dockerfile

```diff
@@ -1,29 +1,30 @@
 FROM node:20-alpine AS builder
+ENV PNPM_HOME="/pnpm"
+ENV PATH="$PNPM_HOME:$PATH"
+RUN corepack enable
 
 WORKDIR /app
-COPY package.json .
-COPY yarn.lock .
 
-RUN yarn install --network-timeout 600000
-
-COPY src/ src
-COPY nginx.conf .
-COPY tsconfig.json .
-COPY vite.config.mts .
 COPY . .
 
 RUN echo `git rev-parse --short HEAD` > health.html
-RUN yarn build
+
+RUN if [ -d "./devtron-fe-lib" ]; then rm -rf packages/devtron-fe-lib; cp -r ./devtron-fe-lib packages/devtron-fe-lib; fi
+
+RUN pnpm i
+RUN pnpm run build:prod
 
 FROM nginx:stable
 
 RUN useradd -ms /bin/bash devtron
-COPY --from=builder /app/dist/ /usr/share/nginx/html
-COPY ./nginx.conf /etc/nginx/nginx.conf
-COPY ./nginx-default.conf /etc/nginx/conf.d/default.conf
+
+COPY --from=builder /app/apps/web/dist/ /usr/share/nginx/html
+COPY --from=builder /app/apps/web/nginx.conf /etc/nginx/nginx.conf
+COPY --from=builder /app/apps/web/nginx-default.conf /etc/nginx/conf.d/default.conf
+
 WORKDIR /usr/share/nginx/html
-COPY --from=builder  /app/./env.sh .
-COPY --from=builder  /app/.env .
+COPY --from=builder  /app/apps/web/./env.sh .
+COPY --from=builder  /app/apps/web/.env .
 COPY --from=builder  /app/health.html .
 
 RUN chown -R devtron:devtron /usr/share/nginx/html
```

Comment:
Cases to be tested:
- Local: OSS / Ent
- OSS / Ent Deployed
- EA OSS / ENT Deployed

Reply:
- Removed the deprecated action yamls
- Dashboard CI